layers/+lang/lua/funcs.el

```diff
@@ -24,7 +24,12 @@
     "sr" 'lua-send-region
     "'" 'lua-show-process-buffer)
   (pcase lua-backend
-    (`lsp-emmy (spacemacs//lua-setup-lsp-emmy))))
+    ('lsp-emmy-lua (spacemacs//lua-setup-lsp-emmy-lua))
+    ('lsp-lua-language-server (spacemacs//lua-setup-lsp-lua-language-server))
+    ('lsp-lua-lsp (spacemacs//lua-setup-lsp-lua-lsp))
+    (_ (if (and (not lua-backend) (configuration-layer/layer-used-p 'lsp))
```

Comment:
This won't work. If `lua-backend` is simply nil, then it uses lua-mode and this is not a hard error.
The hard error is when the user supplied an unexpected value.

See my previous comments for my suggested changes.

Reply:
In new submition https://github.com/syl20bnr/spacemacs/commit/3a56b390a7cd4c23d32c28186af00614bad225b2, it will be error for unexpect value, or try `lsp` when the `lsp-mode` avaiable.